### DISCUSSION TITLE
help with crwlWrap and predicted locations

### DISCUSSION THREAD

**Discussion by MarkJThomas** [bmcclintock/momentuHMM#159] [Q&A] [answered]
Hi,
I hope someone can help! 

Data was recorded at 1 hour intervals (5am-7pm), but it is irregular and has some missing days / hours, wanting to use crwlWrap to plug those gaps and then use HMM to predict behaviours (foraging, resting, flying etc).

Here is a copy of the data:

- Coords were converted to UTM

[vul.csv](https://github.com/bmcclintock/momentuHMM/files/10104921/vul.csv)

when I plot the output I get this:

![Rplot](https://user-images.githubusercontent.com/41441629/204313234-ae0bc6ec-5546-4d35-a1fa-f3d6994f36db.png)

as you can see the upward patterns of predicted points which are very close together (red circles). This isn't normal vulture behaviour and I wondered if there is some code I am missing, or something I can do to get better predicted GPS locations.

TIA

# Fit the crawlWrap function to regularize track
`crw_vul <- crawlWrap(obsData = sub_tot, timeStep = "1 hour",
                      Time.name = "Date_time", coord = c("x", "y"))`

`plot(crw_vul,
     animals = NULL,
     compact = FALSE,
     ask = FALSE,
     crawlPlot = FALSE)`

**Comment by bmcclintock** [comment, marked as answer]
Have you searched [Discussions](https://github.com/bmcclintock/momentuHMM/discussions) for "[crawl](https://github.com/bmcclintock/momentuHMM/discussions?discussions_q=crawl)" and "[crawlWrap](https://github.com/bmcclintock/momentuHMM/discussions?discussions_q=crawlWrap)"? There's lots of similar posts there that might help. There is also a lot of great crawl workshop material [here](https://jmlondon.github.io/crawl-workshop/).

It's hard to diagnose anything without a fully reproducible example, but it looks like the problem areas are longer stretches of missing data. `crawl::crwPredict` won't magically fill these in. If there's no additional information provided about the movement (e.g. by using the `activity` argument to specify periods when it is known that there was no movement), then `crawl::crwPredict` will just draw a straight line between the observations where there are big gaps of missing data (because it's the single best predicted path and doesn't account for uncertainty). See #131 and #81 for potential ways to deal with this.
